core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerPool.scala

```diff
@@ -375,7 +377,8 @@ object ContainerPool {
    * @param memory the amount of memory that has to be freed up
    * @return a list of containers to be removed iff found
    */
-  protected[containerpool] def remove[A](pool: Map[A, ContainerData], memory: ByteSize): List[A] = {
+  @tailrec
+  protected[containerpool] def remove[A](pool: Map[A, ContainerData], memory: ByteSize,toRemove :List[A] = List.empty): List[A] = {
```

Comment:
You'll need to format this with scalafmt to pass CI.

Reply:
Thanks,i will fix it later.